docs/modules/ROOT/pages/how-to/use-devspace.adoc

```diff
@@ -0,0 +1,230 @@
+--
++
+IMPORTANT: Remember to customize the parts marked as `[YOUR_USERNAME]` and `[YOUR_CHOSEN_ZONE]` to your liking (and according to the xref:references/zones.adoc[Zones documentation page]).
+
+. Then change the `dev:replacePods:replaceImage` value, from:
++
+[source,yaml]
+--
+dev:
+  # ...
+  replacePods:
+  - imageSelector: ${IMAGE}
+    replaceImage: loftsh/go:latest
+--
++
+to:
++
+[source,yaml]
+--
+dev:
+  # ...
+  replacePods:
+  - imageSelector: ${IMAGE}
+    replaceImage: registry.gitlab.com/vshn/demos/demo-devspace-custom-image:latest # <1>
+--
+<1> This container image (together with its `Dockerfile`) is available at the https://gitlab.com/vshn/demos/demo-devspace-custom-image[DevSpace in APPUiO Base Image] project.
+
+. Finally, apply a few more `dev:replacePods:patches`, going from this:
++
+[source,yaml]
+--
+# `dev` only applies when you run `devspace dev`
+dev:
+  # ...
+  replacePods:
+    # ...
+    patches:
+    - op: replace
+      path: spec.containers[0].command
+      value:
+      - sleep
+    - op: replace
+      path: spec.containers[0].args
+      value:
+      - "9999999"
+    - op: remove
+      path: spec.containers[0].securityContext
+--
++
+to this:
++
+[source,yaml]
+--
+# `dev` only applies when you run `devspace dev`
+dev:
+  # ...
+  replacePods:
+    # ...
+    patches:
+    - op: replace
+      path: spec.containers[0].command
+      value:
+      - sleep
+    - op: replace
+      path: spec.containers[0].args
+      value:
+      - "9999999"
+    - op: remove
+      path: spec.containers[0].securityContext
+    - op: add # <1>
+      path: spec.containers[0].env
+      value:
+      - name: HOME
+        value: /app
+      - name: GOPATH
+        value: /app/go
+    persistPaths: # <2>
+      - path: /app
+        skipPopulate: true
+--
+<1> New `op` patch, adding environment values.
+<2> New `persistPath` entry.
++
+The `skipPopulate: true` is needed because otherwise an init container tries to copy stuff around, which leads to an `Operation not permitted` error.
+
+=== 6. Launch DevSpace
+
+Launch DevSpace on the terminal:
+
+[source,shell]
+--
+devspace dev
+--
+
+DevSpace deploys your container to the cluster and gives you a terminal prompt; run the `go run main.go` command, and watch your application in action in http://localhost:8080.
+
+Now you can edit any file in your project, and as soon as you save it, DevSpace will automatically sync your files with the pod, increasing your productivity. Just stop the web server with kbd:[CTRL+C], and start it again to see your changes.
+
+Once you're done developing your application, use the following command to deploy it in production:
+
+[source,shell]
+--
+devspace deploy -p production
+--
+
+You can see it in production either by adding an `Ingress` object, or by using the following command:
+
+[source,shell]
+--
+devspace open
+--
+
+and selecting the "via localhost" option.
+
+== Tips & Tricks
+
+Here go some ideas to use DevSpace efficiently:
```

Comment:
Maybe add a tip that the base image could be updated / changed with the OpenShift constraints kept in mind so that it works without the paths and persistPaths patches.

Reply:
I removed the patches, still haven't found how to get rid of the persistPath key

Reply:
Done